### PR TITLE
Add some GC software

### PR DESCRIPTION
**Pull request by jacob-hughes**

### CODE REVIEW

File: softdev.bib [hunk 1]
```diff
@@ -2757,3 +2757,43 @@ @misc{watson21assessing
...
+    year = {2012},
+    author = {Mozilla},
+}
+
+@misc{rustgc,
+    title = {rust-gc: simple tracing (mark and sweep) garbage collector for Rust},
+    howpublished = {\href{https://github.com/manishearth/rust-gc/}{https://github.com/manishearth/rust-gc/}},
+    note = {Accessed: 2022-12-17},
+    year = {2015},
+    author = {Manish Goregaokar},
```
Comment (jacob-hughes) [inline softdev.bib:2774]: This one I'm not too sure about. Manish owns the repo, but it looks like the bulk of contributions came from someone else.

File: softdev.bib [hunk 2]
```diff
@@ -2757,3 +2757,43 @@ @misc{watson21assessing
     month = Sep,
     url = "https://www.capabilitieslimited.co.uk/pdfs/20210917-capltd-cheri-desktop-report-version1-FINAL.pdf"
 }
+
+@misc{servo,
+    title = {The Servo Parallel Browser Engine Project},
+    howpublished = {\href{https://servo.org}{https://servo.org}},
```
Comment (ltratt) [inline softdev.bib:2763]: I notice our other entries are (some? mostly?) `\url` which avoids repeating the URL twice. I guess that's a standard part of pdflatex? [I'm too lazy to remind myself...]

File: softdev.bib [hunk 3]
```diff
@@ -2757,3 +2757,43 @@ @misc{watson21assessing
...
+@misc{shifgrethor,
+    title = {shifgrethor},
+    howpublished = {\href{https://github.com/withoutboats/shifgrethor/}{https://github.com/withoutboats/shifgrethor/}},
+    note = {Accessed: 2022-12-17},
+    year = {2018},
+    author = {withoutboats},
+}
+
+@misc{luster,
+    title = {luster - An experimental Lua VM implemented in pure Rust},
```
Comment (ltratt) [inline softdev.bib:2786]: I can't remember if the outer curlies `{...}` solve this, but you might want to check whether for proper nouns (e.g. "Luster", "Lua") and abbrevations (VM) we need inner curlies around them or not.

File: softdev.bib (hunk 2 above)
Comment (jacob-hughes) [reply softdev.bib:2763]: Ah yes possibly, I can change this if you like. I shamelessly copied this from the octane entry.

File: softdev.bib [hunk 4]
```diff
@@ -2757,3 +2757,43 @@ @misc{watson21assessing
...
+    howpublished = {\href{https://github.com/kyren/luster/}{https://github.com/kyren/luster/}},
+    note = {Accessed: 2022-12-17},
+    year = {2018},
+    author = {kyren},
```
Comment (ltratt) [inline softdev.bib:2790]: I think her name is Catherine West?

File: softdev.bib (hunk 1 above)
Comment (ltratt) [reply softdev.bib:2774]: In such a case you don't even really need an author.

File: softdev.bib (hunk 4 above)
Comment (jacob-hughes) [reply softdev.bib:2790]: I also found this by googling kyren, but wasn't sure if it was right to use this instead as I saw no reference to it from the github profile.

Comment (ltratt) [reply softdev.bib:2790]: We should use real names unless there's a very good reason not to (I think the only reason would be: we can't find out that person's name).

File: softdev.bib (hunk 3 above)
Comment (jacob-hughes) [reply softdev.bib]: I've fixed them here: e039b2b5ac3eb203f1cbb1454874862ccb47fc1d

File: softdev.bib (hunk 4 above)
Comment (jacob-hughes) [reply softdev.bib]: Fixed: 1971e4d7d9b293e9df794079506636667b83ebd7

File: softdev.bib (hunk 2 above)
Comment (jacob-hughes) [reply softdev.bib:2763]: Fixed here: 6bc9582db3f100429727234ecf27249cce9f5014

File: softdev.bib (hunk 1 above)
Comment (jacob-hughes) [reply softdev.bib]: Fixed: a123b4509127c3957bcc7e8d67f6e53d30ad9af4

**Comment by jacob-hughes** [comment]
I think this one is ready for squashing now

File: softdev.bib (hunk 2 above)
Comment (ltratt) [reply softdev.bib:2763]: I *think* only `Servo` should be capitalised. Indeed, I don't see "parallel browser engine project" at all on servo.org?

File: softdev.bib [hunk 5]
```diff
@@ -2759,15 +2759,15 @@ @misc{watson21assessing
...
+    title = "The {Servo} {Parallel} {Browser} {Engine} {Project}",
     howpublished = {\href{https://servo.org}{https://servo.org}},
     note = {Accessed: 2022-12-17},
     year = {2012},
     author = {Mozilla},
 }
 
 @misc{rustgc,
-    title = {rust-gc: simple tracing (mark and sweep) garbage collector for Rust},
+    title = "rust-gc: a simple tracing (mark and sweep) garbage collector for {Rust}",
```
Comment (ltratt) [inline softdev.bib:2770]: If `rust-gc` needs to be lower-case... you'll have to bracket it too!

File: softdev.bib [hunk 6]
```diff
@@ -2783,15 +2783,15 @@ @misc{shifgrethor
 }
 
 @misc{luster,
-    title = {luster - An experimental Lua VM implemented in pure Rust},
+    title = "luster: An experimental {Lua} {VM} implemented in pure {Rust}",
```
Comment (ltratt) [inline softdev.bib:2785]: Same with "luster".

File: softdev.bib (hunk 1 above)
Comment (jacob-hughes) [reply softdev.bib]: Argh! Actually this doesn't work, because prebib complains it's missing an author field.

File: softdev.bib (hunk 5 above)
Comment (jacob-hughes) [reply softdev.bib:2770]: When I do that I get a prebib syntax error which I can't decipher:
```
I was expecting a `,' or a `}'---line 2497 of file out.bib
 :     title = {rust-gc}
 :                      : a simple tracing (mark and sweep) garbage collector for {Rust},
```
Note I am actually including the double quotes "", which seem to be missing from the prebib warning. 

Coincidentally, pdflatex seems to format this as I intended.

Comment (ltratt) [reply softdev.bib:2770]: Sounds like a prebib bug to me? Maybe there's a quick fix to prebib?

Comment (jacob-hughes) [reply softdev.bib]: Ah, it turns out of you have curlys at the beginning and end of a title, you need to wrap the entire title in "" and curlys to keep the bibtex parser happy... Don't ask f7a6f5705ba1f01cc5a2c9365d5e4b0ed54111a8

File: softdev.bib (hunk 2 above)
Comment (jacob-hughes) [reply softdev.bib:2763]: Ah, yeah I took this from their github page. I'll remove it and just leave it as Servo

Comment (jacob-hughes) [reply softdev.bib]: Fixed: 2cbbbff5005e476a234e18147b6495e37d81c14f

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Comment by ltratt** [comment]
Let's try again.

bors r+

**Pull request merged**